### PR TITLE
De-incubate DisableCachingByDefault

### PR DESCRIPTION
**Pull request by lptr**
Fixes: #18269

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/work/DisableCachingByDefault.java [hunk 1]
```diff
@@ -41,7 +40,6 @@
  *
  * @since 7.0
  */
-@Incubating
```
Comment (wolfs) [inline subprojects/core-api/src/main/java/org/gradle/work/DisableCachingByDefault.java:44]: We should add something to the release notes as well, right?

**Review by wolfs — COMMENTED** [review]
LGTM, I think release notes are missing (Promoted section).

File: subprojects/core-api/src/main/java/org/gradle/work/DisableCachingByDefault.java (hunk 1 above)
Comment (lptr) [reply subprojects/core-api/src/main/java/org/gradle/work/DisableCachingByDefault.java:44]: Added.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by lptr** [comment]
@bot-gradle test and merge please.

**Pull request merged**